### DISCUSSION TITLE
How to change deploy path in my Ruby on Rails project?

### DISCUSSION THREAD

**Discussion by reinoutkoninkx** [jgraph/drawio#3473] [Q&A]
Hey,

I have a ruby on rails project in which I want to integrate a draw.io interface, in here I want users to be able to draw stuff and save it etc.
I downloaded the draw.war file of version 21.1. 

I put the extracted files in my public folder, now if I run the rails server, the drawio drawer shows on the homepage ("/").

What I want to happen is to show it only on a dedicated view in my views folder, so I can use controller functionalities to add my own variables and such.

If I move the files from the public folder into a new folder called drawio which is inside the public folder, nothing shows up at all.

To summarize, I want the drawer to only show in my dedicated page which is a view inside the views folder, but still be able to use all the drawio functionalities.

Thanks!

**Comment by davidjgraph** [comment]
None of us have ever used rails, this might not be the best place to ask, since we've no idea what the concept of a view in rails is, for example.

**Reply by reinoutkoninkx** [reply]
It works very similar to any other MVC framework.

Can you explain how to do this in the regular way it was intented for people not using rails?

So just how to change the html page where the actual drawio drawer is deployed?

**Reply by davidjgraph** [reply]
You need to deliver src/main/webapp/index.html , I'm still not sure how your question is a general one. In Apache or Tomcat this is trivial.

**Reply by reinoutkoninkx** [reply]
It's just that I am working on a project which is in ruby on rails for an internship, I have to create a project for the company where I work that decides if I get to graduate, my project is to implement the drawio interface into a current project which is the one made in ruby on rails.

**Reply by reinoutkoninkx** [reply]
I don't know if youre familiar with the MVC principle.
But basically the views can contain any javascript without any problems
![image](https://user-images.githubusercontent.com/99652335/228210112-7ee2cc1c-f6d6-49ed-a8e8-e9b53509061d.png)

This is what the folder looks like, you dont need to have anything in the controller. It is perfectly possible to make it behave like a regular HTML, javascript and css project.

**Reply by davidjgraph** [reply]
I do know the MVC, yes. Unfortunately, I don't understand why you'd need to refactor the project structure into rails in order to use it, that seems rather complex. But I'm probably just old and out of touch with modern tech :). Maybe someone else can help.

**Reply by reinoutkoninkx** [reply]
The current project is an ongoing project, my task is to implement the drawio into one of the pages of the site, so it only appears when I visit that one particular site, right now I put all the files from the draw.war folder into my public folder (a folder where you put your JS and CSS etc), currently when I do this, the drawio drawer is opened on the homepage, I need it to only open on 1 that one specific page.

I have no idea where else to turn for help, any help here is greatly appreciated. If not this, just another way that I can have the drawio drawer on the website. 

I already tried to use an iframe and embed the drawer, which worked, until the point where I needed to change a bit of the interface. When I ran into some problems._
